hermes-management/src/main/java/pl/allegro/tech/hermes/management/config/storage/StorageConfiguration.java

```diff
@@ -172,12 +177,15 @@ OfflineRetransmissionRepository zookeeperOfflineRetransmissionRepository() {
 
     @PostConstruct
     public void init() {
+        logger.info("Before ensuring init path exists");
         ensureInitPathExists();
+        logger.info("After ensuring init path exists");
     }
 
     private void ensureInitPathExists() {
         ZookeeperClientManager clientManager = clientManager();
         for (ZookeeperClient client : clientManager.getClients()) {
+            logger.info("Ensuring that path exists for client: " + client);
             client.ensurePathExists(zookeeperPaths().groupsPath());
```

Comment:
One more thing - `client.ensurePathExists` method verifies if path exists, and if not, it tries to create it. This is the method itself:

```java
public void ensurePathExists(String path) {
    try {
        if(curatorFramework.checkExists().forPath(path) ==  null) {
            curatorFramework.create().creatingParentsIfNeeded().forPath(path);
        }
    } catch (Exception e) {
        throw new InternalProcessingException("Could not ensure existence of path: " + path);
    }
}
```

But if there is an error during evaluation of`curatorFramework.checkExists` or `curatorFramework.create`, `new InternalProcessingException` is thrown, **but there is no error logged**. 

Can I recommend you to add more logs in this `ensurePathExists` method? For example:

```java
public void ensurePathExists(String path) {
    try {
        if(curatorFramework.checkExists().forPath(path) ==  null) {
            logger.info("Creating path: {} in Zookeeper: {}", path, datacenterName);
            curatorFramework.create().creatingParentsIfNeeded().forPath(path);
        }
    } catch (Exception e) {
        logger.error("Error when creating path: {} in Zookeeper: {}", path, datacenterName, e);
        throw new InternalProcessingException("Could not ensure existence of path: " + path);
    }
}
```